src/ChorizoServer.cs

```diff
@@ -9,20 +11,24 @@ public class Chorizo
         private IServerStatus Status { get; }
         private ISocketMachine SocketMachine { get; }
         private IChorizoProtocolConnectionHandler ProtocolConnectionHandler { get; }
+        private IMiniLogger Logger { get; }
         private ServerConfig Config { get; }
 
         public Chorizo(
             int port = 8000,
             string protocol = "HTTP",
+            string mode = "dev",
             IServerStatus serverStatus = null,
             ISocketMachine socketMachine = null,
-            IChorizoProtocolConnectionHandler protocolConnectionHandler = null
+            IChorizoProtocolConnectionHandler protocolConnectionHandler = null,
+            IMiniLogger logger = null
```

Comment:
This constructor now has 7 inputs. I wonder if we can think of some ways to reduce that